src/redux/books/books.js

```diff
@@ -14,12 +20,14 @@ const booksReducer = (state = [], action) => {
   }
 };
 
-const addBook = () => ({
+const addBook = (payload) => ({
   type: BOOK_ADDED,
+  payload,
 });
 
-const removeBook = () => ({
+const removeBook = (id) => ({
   type: BOOK_REMOVED,
+  payload: { id },
 });
```

Comment:
- [ ] There's an error with the removeBook functionality. If you pay close attention, you're sending an object like this as payload: `{ id: id }` and your reducer is filtering by the condition of the `book.id` being different than this payload, for instance: `book.id !== { id: id }` This condition never returns true, `book.id` is a string, and `action.payload` is an object.

💡 To solve this, you could just remove the curly brackets `{ }` from the payload, like this:

```js
const removeBook = (id) => ({
  type: BOOK_REMOVED,
  payload: id,
});
```

Now, you're checking that the `book.id !== id` which is what you want. 🤓 🔧

Reply:
@anagudelogu Great review! Thank you very much for the explanation... I will fix it